purple/client/callbacks.py

```diff
@@ -0,0 +1,89 @@
+from typing_extensions import Protocol
+from typing import Callable
+from typing import Any
+
+import purple
+
+
+class CallbackSignalConversationReceivedImMsg(Protocol):
```

Comment:
Why not `Callable[[/**/], None]`?

Reply:
I find Protocols much easier to read, especially when using kwargs.

The Callable syntax for this is ugly:
```
F = Callable[[int,  # Or Arg(int)
              Arg(int, 'b'),
              DefaultArg(int, 'c'),
              VarArg(int),
              NamedArg(int, 'd'),
              DefaultNamedArg(int, 'e'),
              KwArg(int)],
             int]
```

Callbacks are actually a documented use case for Protocols:
 - https://mypy.readthedocs.io/en/latest/protocols.html#callback-protocols

It also says:
> Callback protocols and Callable types can be used interchangeably.

Reply:
Sounds good then!